pkg/resource/parameter_group/hooks.go

```diff
@@ -158,6 +159,119 @@ func (rm *resourceManager) resetAllParameters(
 	return &resource{ko}, nil
 }
 
+// customUpdate overrides sdkUpdate by custom logic
+func (rm *resourceManager) customUpdate(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) (updated *resource, err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.customUpdate")
+	defer func() {
+		exit(err)
+	}()
+	if delta.DifferentAt("Spec.Tags") {
+		err = rm.updateTags(ctx, desired, latest)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if delta.DifferentAt("Spec.ParameterNameValues") {
+		ko, err := rm.resetParameterGroup(ctx, desired, latest)
+
+		if ko != nil || err != nil {
+			return ko, err
+		}
+	}
+
+	if !delta.DifferentExcept("Spec.Tags") {
+		return desired, nil
+	}
+
+	inputs := rm.updateRequestPayload(desired)
+
+	var resp *svcsdk.UpdateParameterGroupOutput
+	_ = resp
+
+	// Update 20 parameters each time
+	for _, input := range inputs {
+		resp, err = rm.sdkapi.UpdateParameterGroupWithContext(ctx, input)
+		rm.metrics.RecordAPICall("UPDATE", "UpdateParameterGroup", err)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Merge in the information we read from the API call above to the copy of
+	// the original Kubernetes object we passed to the function
+	ko := desired.ko.DeepCopy()
+
+	if ko.Status.ACKResourceMetadata == nil {
+		ko.Status.ACKResourceMetadata = &ackv1alpha1.ResourceMetadata{}
+	}
+	if resp.ParameterGroup.ARN != nil {
+		arn := ackv1alpha1.AWSResourceName(*resp.ParameterGroup.ARN)
+		ko.Status.ACKResourceMetadata.ARN = &arn
+	}
+	if resp.ParameterGroup.Description != nil {
+		ko.Spec.Description = resp.ParameterGroup.Description
+	} else {
+		ko.Spec.Description = nil
+	}
+	if resp.ParameterGroup.Family != nil {
+		ko.Spec.Family = resp.ParameterGroup.Family
+	} else {
+		ko.Spec.Family = nil
+	}
+	if resp.ParameterGroup.Name != nil {
+		ko.Spec.Name = resp.ParameterGroup.Name
+	} else {
+		ko.Spec.Name = nil
+	}
+
+	rm.setStatusDefaults(ko)
+	ko, err = rm.setParameters(ctx, ko)
+
+	if err != nil {
+		return nil, err
+	}
+	return &resource{ko}, nil
+}
+
+// updateRequestPayload returns an array of SDK-specific struct for the HTTP request
+// payload of the Update API call for the resource
+// each element in return value has maximum 20 parameters
+func (rm *resourceManager) updateRequestPayload(
+	desired *resource,
+) ([]*svcsdk.UpdateParameterGroupInput) {
+	inputs := []*svcsdk.UpdateParameterGroupInput{}
+	parameterNameValues := lo.Chunk[*svcapitypes.ParameterNameValue](desired.ko.Spec.ParameterNameValues, 20)
+
+	for _, iter := range parameterNameValues {
```

Comment:
nit: This could be even simpler if you did `return lo.Chunk(desired.ko.Spec.ParameterNameValues, 20).Map(...)`. You could make this into a single statement

Reply:
I use `lo.Map` to replace the outer `for` loop.